pkg/telemetry/metrics_helper.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package telemetry
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/cli"
+	configlib "github.com/vmware-tanzu/tanzu-plugin-runtime/config"
+	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
+)
+
+func getEndpointSHA(plugin *cli.PluginInfo) string {
+	cfg, err := configlib.GetClientConfig()
+	if err != nil {
+		return ""
+	}
+	curCtxMap, err := cfg.GetAllCurrentContextsMap()
+	if err != nil || curCtxMap == nil {
+		return ""
+	}
+
+	if plugin.Target == configtypes.TargetK8s {
+		return getCurrentK8SContextSHA(curCtxMap)
+	} else if plugin.Target == configtypes.TargetTMC {
+		return getCurrentTMCContextSHA(curCtxMap)
+	}
+
+	return ""
+}
+
+// getCurrentK8SContextSHA returns SHA1 of the complete context
+func getCurrentK8SContextSHA(curCtx map[configtypes.Target]*configtypes.Context) string {
+	ctx, exists := curCtx[configtypes.TargetK8s]
+	if exists {
+		ctxBytes, _ := json.Marshal(ctx)
+		// #nosec using the sha1 just for masking the value and not intended for signatures/certs
```

Comment:
sha256?

Reply:
Removing this comment as it(#nosec) is not required after i changes SHA1 to SHA256.